cmd/noms/migrate/migrate.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package migrate
+
+import (
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/types"
+	v7types "github.com/attic-labs/noms/go/types"
+)
+
+func Value(source v7types.Value, sourceStore v7types.ValueReadWriter, sinkStore types.ValueReadWriter) (dest types.Value, err error) {
+	switch source := source.(type) {
+	case v7types.Bool:
+		return types.Bool(bool(source)), nil
+	case v7types.Number:
+		return types.Number(float64(source)), nil
+	case v7types.String:
+		return types.String(string(source)), nil
+	case v7types.Blob:
+		return types.NewStreamingBlob(source.Reader(), sourceStore), nil
+	case v7types.List:
+		vc := make(chan types.Value, 1024)
+		lc := types.NewStreamingList(sinkStore, vc)
+		for i := uint64(0); i < source.Len(); i++ {
+			var nv types.Value
+			nv, err = Value(source.Get(i), sourceStore, sinkStore)
+			if err != nil {
+				break
+			}
+			vc <- nv
+		}
+		close(vc)
```

Comment:
Cannot use `defer close(vc)` since close is the signal to the streaming builders that we are done building. I could add a closure to simulate a try/finally but it gets ugly. If something goes wrong in here we exit the inner loop and close it which leads to a partial collection but we keep the error.